### DISCUSSION TITLE
Allow multiple celebration definitions within a calendar

### DISCUSSION THREAD

**Discussion by tukusejssirs** [romcal/romcal#307] [Ideas]
I think we should allow multiple celebration definitions within a calendar.

Reasons:
- it seems very probable that the Vietnamese Martyrs are celebrated on two different dates in Vietnam;
- in Slovakia, we celebrate Sts Cyril and Methodius on 5 July, but we also can celebrate them on 14 February as a votive Mass;
- when romcal will support votive Masses from the Martyrology, surely there’ll be some ‘duplicate’ celebration options throughout the year in a calendar.

Sure, the votive Masses are not yet supported by romcal. Also, the votive Masses are not part of any calendar proper, but they might be part of some ordos (e.g. in Slovakia, our official ordo does contain several votive Masses).

---

Related: #110, #255 and #259. See also [this comment of mine](https://github.com/romcal/romcal/pull/259#issuecomment-774687684).

**Comment by emagnier** [comment]
About **Votive Masses**, from the GIRM, §355 c:

> On the weekdays in Ordinary Time, it is possible to choose either a weekday Mass, or the Mass of an optional memorial which happens to occur on that day, or the Mass of any Saint listed in the Martyrology for that day, or a Mass for Various Needs, or a **Votive Mass**.

So a Votive Mass isn't a liturgical day in its own right, but an alternative mass within the `LiturgicalDay` of the weekday.
An optional memorial taken for a day is a bit different, since specific prayers and readings could also be taken in the liturgy of the hours.

So in my point of view, a Votive Mass should not be a `LiturgicalDay` object, but a new type of object that will be part of a `LiturgicalDay`, and outputted only if the conditions are met.

Consequently, a Votive Mass should not be defined the same way as a Liturgical Day. And indeed, a Votive Mass can occur multiple time in a year (the opposite of a Liturgical Day, which I believe must remain unique in a year).

**Reply by emagnier** [reply]
I also want to emphasize that romcal isn't managing celebrations, but liturgical days.

Liturgical Days are of course including celebrations (masses, liturgy of the hours...), although no specific data about celebrations are currently outputted by romcal.

**Comment by emagnier** [comment]
> it seems very probable that the Vietnamese Martyrs are celebrated on two different dates in Vietnam;

When a saint is celebrated twice in a calendar, the second date is generally a bit different (e.g.: translation of the relics), so the 2 dates are not exactly the same `LiturgicalDay` definitions.

Another case is the _Dedication of Consecrated Churches whose date of Consecration is unknown_, that can be proposed on 2 different dates on some particular calendars. But in the same way here, it is only one of the 2 dates which must be chosen. Actually this is solved by duplicating this `LiturgicalDay` definition, but applying a different key. Not too DRY, but this is the only official case I know where a same Liturgical Day could be proposed twice in a calendar, so I think it's reasonable to keep the way they are currently defined.

To come back to the _Vietnamese Martyrs_, I'll wait for an official source before considering modifying the definition system of romcal. I bet that if there are 2 proposed dates, this is to choose one or the other, similarly to the _Dedication of Consecrated Churches_, or else that the second date is different (e.g.: translation of the relics).
